### PR TITLE
experiments: add options to include/exclude columns in `exp show`

### PR DESCRIPTION
**Pull request by pmrowla**
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

Related to #4346

* Adds `--include-metrics/--exclude-metrics` and `--include-params/--exclude-params` options to `dvc exp show`.
* Options take comma-separated list of metrics/params columns to include or exclude from the output table.
* If `--include...` is used, only columns specified in the list will be displayed. If `--exclude...` is used, all columns except those specified in the list will be displayed.
* Matches nested metrics/params, so `--include-params featurize` will match `featurize.max_features`, `featurize.ngrams`, etc
* Params and metrics columns lists are handled separately, so `dvc exp show --include-params=foo` will display all metrics columns plus the single param column `foo`.

### CODE REVIEW

**Comment by pmrowla** [comment]
```

example-get-started git:cli-params  py:dvc ❯ dvc exp show --include-params featurize.max_features,featurize.ngrams --no-pager                              ⏎
┏━━━━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━┓
┃ Experiment  ┃     auc ┃ featurize.max_features ┃ featurize.ngrams ┃
┡━━━━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━┩
│ workspace   │ 0.61314 │ 1500                   │ 2                │
│ acd59eb     │ 0.61314 │ 1500                   │ 2                │
│ ├── be1e801 │  0.5125 │ 500                    │ 3                │
│ ├── 1dd825b │ 0.51676 │ 500                    │ 2                │
│ └── eb12d21 │ 0.54175 │ 500                    │ 1                │
└─────────────┴─────────┴────────────────────────┴──────────────────┘
```

**Comment by pmrowla** [comment]
marking this WIP, will add support for filtering on higher level (i.e. `--include-params=featurize` would match `featurize.max_features`, `featurize.ngrams`, etc)

File: dvc/command/experiments.py [hunk 1]
```diff
@@ -2,14 +2,57 @@
...
+                f" --exclude-{label}"
+            )
+
+    names = {tuple(name.split(".")) for name in names}
+
+    def _filter(filters, update_func):
+        filters = {tuple(name.split(".")) for name in filters}
+        for length, groups in groupby(filters, len):
+            for group in groups:
+                matches = [name for name in names if name[:length] == group]
```
Comment (pared) [inline dvc/command/experiments.py:36]: If we are going to the extent of matching on high level, maybe we could consider support for expressions?
Just an idea, and not for this change, I am pretty sure that parsing from the command line will be a headache if we were to go this way.

Comment (pmrowla) [reply dvc/command/experiments.py:36]: Yeah I was thinking about this too. Support for more complex expressions seems like a nice to have we could consider adding down the line, but would be overkill for now.

**Pull request merged**